docs/system_overview/configuration.md

```diff
@@ -14,244 +14,640 @@ This scrambled password file is saved after an `iinit` is run. If this file does
+        // ready for execution.
+        "delay_server_sleep_time_in_seconds": 30,
+
+        // (Optional)
+        // Contains settings for controlling the behavior of the internal DNS cache.
+        "dns_cache": {
+            // The maximum amount of shared memory allocated to the DNS cache.
+            "shared_memory_size_in_bytes": 5000000,
+
+            // The amount of time an entry stays valid before being evicted from the cache.
+            "eviction_age_in_seconds": 3600
+        },
 
-  - `db_odbc_type` (required) - The ODBC type, usually 'unix'
+        // (Optional)
+        // Contains settings for controlling the behavior of the internal Hostname cache.
+        "hostname_cache": {
+            // The maximum amount of shared memory allocated to the Hostname cache.
+            "shared_memory_size_in_bytes": 2500000,
 
-  - `db_password` (required) - The password for the `db_username` to connect to the `db_name`
+            // The amount of time an entry stays valid before being evicted from the cache.
+            "eviction_age_in_seconds": 3600
+        },
 
-  - `db_port` (required) - The port on which the database server is listening
+        // TODO
+        "maximum_size_for_single_buffer_in_megabytes": 32,
+
+        // (Optional)
+        // The maximum number of bytes available to the delay queue.
+        // When set to 0, the delay server will use as much memory as it needs to hold queued rules.
+        "maximum_size_of_delay_queue_in_bytes": 0,
+
+        // TODO
+        "maximum_temporary_password_lifetime_in_seconds": 1000,
+
+        // (Optional)
+        // The number of delay rules allowed to execute simultaneously.
+        "number_of_concurrent_delay_rule_executors": 4,
+
+        // (Optional)
+        // The amount of time the stacktrace file processing thread sleeps before files are
+        // processed and written to the log file.
+        "stacktrace_file_processor_sleep_time_in_seconds": 10,
+
+        // TODO
+        "transfer_buffer_size_for_parallel_transfer_in_megabytes": 4,
+
+        // TODO
+        "transfer_chunk_size_for_parallel_transfer_in_megabytes": 40
+    },
+
+    // An array of fully qualified domain names of this Zone's catalog service provider.
+    "catalog_provider_hosts": [],
+
+    // The role of this server.
+    // The following values are supported: provider, consumer
+    "catalog_service_role": "",
+
+    // Controls which API operations are accessible to non-rodsadmin users.
+    //
+    // When set to "enforce", the server will expose a subset of operations that can be executed by any user.
+    // Administrative operations will require rodsadmin level privileges. This option does not apply to
+    // rodsadmin users.
+    //
+    // When set to "disabled", the server will operate without consulting the allowlist. This is equivalent
+    // to how server behaved prior to iRODS 4.3.0.
+    //
+    // The following values are supported: enforce, disabled
```

Comment:
consider ...

> The following two values are supported:
> 
>  - When set to "enforce" ...
> 
>  - When set to "disabled" ...

Reply:
Changed to the following to keep things inline with how other properties define what range of acceptable values.

Basically, I moved the line declaring the values above the lines describing what each value means.

I feel this helps the reader in two ways. (1) They are informed of the accepted values. (2) By listing the options up-front, they don't have to look over the entire list to see what is valid. They can pick and choose which one they are interested in. This should probably be the rule for all settings.
```
Controls which API operations are accessible to non-rodsadmin users.

The following values are supported: enforce, disabled

When set to "enforce", the server will expose a subset of operations that can be executed by any user.
Administrative operations will require rodsadmin level privileges. This option does not apply to rodsadmin users.

When set to "disabled", the server will operate without consulting the allowlist. This is equivalent
to how the server behaved prior to iRODS 4.3.0.
```

Reply:
Yeps, looks good.